packages/widgets/src/lib/components/InferenceWidget/shared/WidgetWrapper/WidgetWrapper.svelte

```diff
@@ -35,6 +35,10 @@
 			if (modelLoadInfo?.state === "TooBig") {
 				updateWidgetState(model.id, "isDisabled", true);
 			}
+
+			if (!$tgiSupportedModels) {
+				$tgiSupportedModels = await getTgiSupportedModels(apiUrl);
+			}
```

Comment:
Do we need it for every widgets? 

If yes, that's ok to initialize it here, 
otherwise, you can do something like this : 

in store.js 
```ts
import { get, writable } from 'svelte/store'; 

const tgiSupportedModels = writable<Set<string> | undefined>(undefined);

export async function getTgiSupportedModels() {
     if (!get(tgiSupportedModels)) { 
        const response = await fetch(`${url}/framework/text-generation-inference`);
	const output = await response.json();
	if (response.ok) {
		tgiSupportedModels.set(new Set(
			(output as { model_id: string; task: string }[])
				.filter(({ task }) => task === "text-generation")
				.map(({ model_id }) => model_id)
		));
	}
        return tgiSupportedModels;
}
``` 

so inside widgets, you always get the store using `const $tgiSupportedModels = getTgiSupportedModels()` and it fetches the data only when needed.

Reply:
so nice! thank you!